### PR TITLE
Build OpenWrt packages in main pipeline

### PR DESCRIPTION
**Pull request by robaerd**
Part of #3898

## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by robaerd** [comment]
Somehow overwriting the makefile variables of the openwrt elektra makefile was not possible, making it necessary to create a script that uses `sed` to remove the configuration for releases and use the current state of the master branch for the package builds. 
More precisely, overwriting the variables with e.g. `make PKG_SOURCE_PROTO=git <MORE VARIABLES> package/feeds/packages/elektra/` was possible, but then some `make` processes terminated due to too many arguments. I could not figure out the exact cause of this, but IMO the approach with the script would be even better, as unsetting a value is not possible with command line args (only setting it to an empty value).

File: scripts/jenkins/Jenkinsfile.release [hunk 1]
```diff
@@ -677,6 +686,26 @@ def buildRelease(stageName, image, packageRevision='1',
...
+ * @param image which docker image should be used
+ */
+def buildOpenwrtPackage(image) {
+  def stageName = 'build-openwrt-package'
+  return [(stageName): {
+    stage(stageName) {
+      withDockerEnv(image, [DockerOpts.MOUNT_MIRROR]) {
+        checkout scm
+        sh "./scripts/release/prepare-openwrt-makefile.sh /home/build/openwrt/package/feeds/packages/elektra/Makefile"
+        sh "cd /home/build/openwrt && make package/feeds/packages/elektra/compile V=sc"
```
Comment (markus2330) [inline scripts/jenkins/Jenkinsfile.release:702]: Does this maybe create too many logs? What about first try as fast as possible without verbosity, if it fails without concurrency and full verbosity, e.g.:

```suggestion
        sh "cd /home/build/openwrt && make package/feeds/packages/elektra/compile || make -j1 V=sc package/feeds/packages/elektra/compile"
```

see also #3904

**Review by markus2330 — APPROVED** [review]
Very good job! I also like the script. Thank you for adding this :sparkling_heart:

**Comment by robaerd** [comment]
I wasn't successful in creating cross-compiled packages for ARM from an x86_64 architecture.
I believe the issue is that the other packages Elektra depends on, do not get cross-compiled in the desired architecture. 

Is cross compiling from a x86_64 architecture even the right approach for such packages? We wouldn't be able to test packages for different architectures anyway.

The community maintained packages (https://github.com/openwrt/packages) also do not cross-compile packages, instead they use GitHub Actions to build and test these packages (including Elektra) on different architectures.

Maybe we should also build+test the Elektra openwrt package on GitHub Actions with the necessary architectures instead of Jenkins?

**Comment by robaerd** [comment]
> Nothing regarding OpenWRT is built in the Jenkins pipeline from what I see?

I moved the OpenWRT build from the release pipeline into the main pipeline. The OpenWRT packages should now be built on every master commit.

**Comment by markus2330** [comment]
Thank you, great work!

**Pull request merged**